### PR TITLE
Update round-tripping of numbers

### PR DESCRIPTION
**Pull request by gkellogg**
to consider non-fractional numbers >= 0e21 and add test.

### CODE REVIEW

**Comment by gkellogg** [comment]
Fixes #79.

File: index.html [hunk 1]
```diff
@@ -4737,8 +4737,10 @@ <h2>Data Round Tripping</h2>
       representation as defined in [[XMLSCHEMA11-2]].</p>
 
     <p>The <a>canonical lexical form</a> of an <em>integer</em>, i.e., a
-      <a>number</a> with no non-zero fractional part or a <a>number</a>
-      coerced to <code>xsd:integer</code>, is a finite-length sequence of decimal
+      <a>number</a> with no non-zero fractional part
+      <span class="changed">and an exponent less than or equal to <code>21</code></code>,</span>
```
Comment (dlongley) [inline index.html]:
```suggestion
      <span class="changed">and an exponent less than or equal to <code>21</code>,</span>
```

File: index.html [hunk 2]
```diff
@@ -4737,8 +4737,10 @@ <h2>Data Round Tripping</h2>
       representation as defined in [[XMLSCHEMA11-2]].</p>
 
     <p>The <a>canonical lexical form</a> of an <em>integer</em>, i.e., a
-      <a>number</a> with no non-zero fractional part or a <a>number</a>
-      coerced to <code>xsd:integer</code>, is a finite-length sequence of decimal
+      <a>number</a> with no non-zero fractional part
+      <span class="changed">and an exponent less than or equal to <code>21</code>,</span>
```
Comment (dlongley) [inline index.html]: I think ECMAScript says `< 1e21` (*not* `<= 1e21`) https://tc39.github.io/ecma262/#sec-number.prototype.tofixed
```suggestion
      <span class="changed">and an exponent less than <code>21</code>,</span>
```

Comment (gkellogg) [reply index.html]: Yup, mis-read.

**Comment by pchampin** [comment]
Shouldn't the 'Data model' of the Syntax document be updated accordingly?
It says:
> a number (numbers with a non-zero fractional part, i.e., the result of a modulo‑1 operation, are interpreted as typed values with type xsd:double, all other numbers are interpreted as typed values with type xsd:integer)

**Pull request merged**